### PR TITLE
Filter tomatoes by creation date

### PR DESCRIPTION
**Pull request by potomak**
Closes https://github.com/tomatoes-app/tomatoes/issues/241.

### CODE REVIEW

File: app/controllers/api/tomatoes_controller.rb [hunk 1]
```diff
@@ -39,6 +42,14 @@ def destroy
 
     private
 
+    def from
+      @from ||= Time.zone.parse(params[:from].to_s)
```
Comment (bugant) [inline app/controllers/api/tomatoes_controller.rb]: How does it handle nil?

Comment (dalpo) [inline app/controllers/api/tomatoes_controller.rb]: It should work! :)

```ruby
2.3.3 :001 > nil.to_s
 => ""
2.3.3 :002 > Time.zone.parse ''
 => nil 
```

File: app/controllers/api/tomatoes_controller.rb [hunk 2]
```diff
@@ -4,7 +4,10 @@ class TomatoesController < BaseController
     before_action :find_tomato, only: [:show, :update, :destroy]
 
     def index
-      @tomatoes = current_user.tomatoes.order_by([[:created_at, :desc], [:_id, :desc]]).page params[:page]
+      @tomatoes = current_user.tomatoes
+      @tomatoes = @tomatoes.after(from) if from
+      @tomatoes = @tomatoes.before(to) if to
+      @tomatoes = @tomatoes.order_by([[:created_at, :desc], [:_id, :desc]]).page params[:page]
```
Comment (dalpo) [inline app/controllers/api/tomatoes_controller.rb]: Why don't you create a scope for sorting tomatoes?

File: app/controllers/tomatoes_controller.rb [hunk 3]
```diff
@@ -75,7 +75,7 @@ def create
       if @tomato.save
         format.js do
           @highlight      = @tomato
-          @tomatoes       = current_user.tomatoes.after(Time.zone.now.beginning_of_day)
+          @tomatoes       = current_user.tomatoes.after(Time.zone.now.beginning_of_day).order_by([[:created_at, :desc]])
```
Comment (dalpo) [inline app/controllers/tomatoes_controller.rb]: If you create a scope for sorting, you can reuse it here too :)

File: app/controllers/api/tomatoes_controller.rb (hunk 1 above)
Comment (potomak) [inline app/controllers/api/tomatoes_controller.rb]: Yes, that's how. `#parse` should work on any string, it returns `nil` in case of parsing failure.

**Comment by potomak** [comment]
Yes @dalpo, I wanted to create such a scope, but I didn't find a good name for it. `sorted_desc` is not descriptive enough. I thought of `reverse_chronological_order`, but it's too long. I though of `latest_first` or `freshest_first`, but they have *first* in them that seem like a selection of the first element. I thought of `created_at_desc`, but it wouldn't include the second sorting condition on `_id`.

**Comment by dalpo** [comment]
I like sorted by `#freshest` tomatoes :smile: 
...but It's not a good name.

Maybe simply: `Tomato.descendants`? I think it's self explanatory enough.

**Comment by potomak** [comment]
@dalpo naming is too hard! I'll leave this task for another PR. I created a proposal where we can discuss on an appropriate name for the scope: https://github.com/tomatoes-app/tomatoes/issues/256.

**Pull request merged**